hack/generator/pkg/codegen/pipeline/testdata/TestInjectConvertibleStatusInterface/microsoft.person-v20200101.golden

```diff
@@ -0,0 +1,186 @@
+// Code generated by azure-service-operator-codegen. DO NOT EDIT.
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+package v20200101
+
+import (
+	"github.com/Azure/azure-service-operator/hack/generated/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/testing/microsoft.person/v20200101storage"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+type Person struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              Person_Spec   `json:"spec,omitempty"`
+	Status            Person_Status `json:"status,omitempty"`
```

Comment:
I tried combining the two stages together, but there are enough differences that the resulting stage factory was hard to read; I've therefore left it the way it was.

The two interfaces need to be separate because we want to catch any case where the wrong interface is passed around as early as possible. This is less important right now, but as I write the code to integrate into the generic controller, having the compiler catch any issues will be most useful.

I understand (and agree with) the sentiment that the golden files can be hard to read - but I'm not sure combining them actually makes things easier to understand. I've tried to somewhat mitigate this by adding comments detailing what needs to be checked in each golden file.

Reply:
Does it make sense to generate types with just spec or just status having the interface impled, but not the other one? Or is the expectation that these two stages are always used together?